python/cudf/cudf/utils/cudautils.py

```diff
@@ -608,6 +608,44 @@ def apply_round(data, decimal):
     return output_dary
 
 
+@cuda.jit
+def gpu_clip_inplace(in_col, lower, upper):
+    i = cuda.grid(1)
+
+    if i < in_col.size:
+        if in_col[i] < lower:
+            in_col[i] = lower
+        elif in_col[i] > upper:
+            in_col[i] = upper
+
+
+@cuda.jit
+def gpu_clip_not_inplace(in_col, lower, upper, out_col):
+    i = cuda.grid(1)
+
+    if i < in_col.size:
+        if in_col[i] < lower:
+            out_col[i] = lower
+        elif in_col[i] > upper:
+            out_col[i] = upper
+        else:
+            out_col[i] = in_col[i]
+
+
+def apply_clip(data, lower, upper, inplace):
+    if not inplace:
+        output_dary = rmm.device_array_like(data)
+        if data.size > 0:
+            gpu_clip_not_inplace.forall(data.size)(
+                data, lower, upper, output_dary
+            )
+        return output_dary
+    else:
+        if data.size > 0:
+            gpu_clip_inplace.forall(data.size)(data, lower, upper)
+        return data
```

Comment:
This can be reworked to have `output_dary` be data depending on the `inplace` parameter

Reply:
Aha, something like this, right?
``` 
output_dary = data if inplace else rmm.device_array_like(data)
```